book/src/dev/rfcs/0006-contextual-difficulty.md

```diff
@@ -0,0 +1,813 @@
+  block's difficulty field). On testnet, if a long time elapses between blocks,
+  the difficulty filter also allows minimum-difficulty blocks.
+
+* **block work**: The approximate amount of work required for a miner to generate
+  a block hash that passes the difficulty filter. The number of block header
+  attempts and the mining time are proportional to the work value. Numerically
+  higher work values represent longer processing times.
+
+* **averaging window**: The 17 most recent blocks in the relevant chain.
+
+* **median block span**: The 11 most recent blocks from a chosen tip, typically
+  the relevant tip.
+
+* **target spacing**: 150 seconds per block before Blossom activation, 75 seconds
+  per block from Blossom activation onwards.
+
+* **mean target difficulty**: The arithmetic mean of the difficulty thresholds
+  of the blocks in the averaging window.
+
+* **median timespan**: The average number of seconds taken to generate the blocks
+  in the averaging window. Calculated using the difference of median block spans
+  in and after the averaging window, then damped and bounded.
+
+* **target timespan**: The target spacing for an averaging window's worth of
+  blocks.
+
+Consensus:
+* **consensus rule:** A protocol rule which all nodes must apply consistently,
+                      so they can converge on the same chain fork.
+
+* **structural/semantic/contextual verification**: as defined in [RFC2].
+
+State:
+* **block chain**: A sequence of valid blocks linked by inclusion of the
+  previous block hash in the subsequent block. Chains are rooted at the
+  genesis block and extend to a tip.
+
+* **relevant chain**: The relevant chain for a block starts at the previous
+  block, and extends back to genesis.
+
+* **relevant tip**: The tip of the relevant chain.
+
+* **non-finalized state**: State data corresponding to blocks above the reorg
+  limit. This data can change in the event of a chain reorg.
+
+* **finalized state**: State data corresponding to blocks below the reorg
+  limit. This data cannot change in the event of a chain reorg.
+
+* **non-finalized tips**: The highest blocks in each non-finalized chain. These
+  tips might be at different heights.
+
+* **finalized tip**: The highest block in the finalized state. The tip of the best
+  chain is usually 100 blocks (the reorg limit) above the finalized tip. But it can
+  be lower during the initial sync, and after a chain reorganization, if the new
+  best chain is at a lower height.
+
+# Guide-level explanation
+[guide-level-explanation]: #guide-level-explanation
+
+The difficulty threshold for the next block is calculated using the difficulty
+thresholds and times of recent blocks. Zcash uses the most recent 28 blocks in
+the **relevant chain** in its difficulty adjustment calculations.
+
+The difficulty adjustment calculations adjust the **mean target difficulty**,
+based on the difference between the **median timespan** and the
+**target timespan**.
```

Comment:
TODO:
- [ ] cut detail from the guide-level explanation - turn it into a summary
- [ ] add a short note explaining how the difficulty filter and difficulty adjustment interact

Reply:
I moved "cut detail from the guide-level explanation - turn it into a summary" into the tracking issue, because it's more work than I expected.